scripts/restore

```diff
@@ -1,58 +1,95 @@
 #!/bin/bash
 
-# Exit on command errors and treat unset variables as an error
+#=================================================
+# GENERIC START
+#=================================================
+# MANAGE SCRIPT FAILURE
+#=================================================
+
+# Exit on command errors and treat access to unset variables as an error
 set -eu
 
-# Get multi-instances specific variables
+#=================================================
+# IMPORT GENERIC HELPERS
+#=================================================
+
+if [ ! -e _common.sh ]; then
+	# Get the _common.sh file if it's not in the current directory
+	cp ../settings/scripts/_common.sh ./_common.sh
+	chmod a+rx _common.sh
+fi
+source _common.sh
+source /usr/share/yunohost/helpers
+
+#=================================================
+# LOAD SETTINGS
+#=================================================
+
 app=$YNH_APP_INSTANCE_NAME
 
-# Set app specific variables
-dbname=$app
-dbuser=$app
-
-# Source app helpers
-. /usr/share/yunohost/helpers
-
-# Retrieve old app settings
-domain=$(ynh_app_setting_get "$app" domain)
-path=$(ynh_app_setting_get "$app" path)
-dbpass=$(ynh_app_setting_get "$app" mysqlpwd)
-
-# Check domain/path availability
-sudo yunohost app checkurl "${domain}${path}" -a "$app" \
-  || exit 1
-
-# Check destination directory
-DESTDIR="/var/www/$app"
-[[ -d $DESTDIR ]] && ynh_die \
-"The destination directory '$DESTDIR' already exists.\
- You should safely delete it before restoring this app."
-
-# Check configuration files
-nginx_conf="/etc/nginx/conf.d/${domain}.d/${app}.conf"
-[[ -f $nginx_conf ]] && ynh_die \
-"The NGINX configuration already exists at '${nginx_conf}'.
- You should safely delete it before restoring this app."
-phpfpm_conf="/etc/php5/fpm/pool.d/${app}.conf"
-[[ -f $phpfpm_conf ]] && ynh_die \
-"The PHP FPM configuration already exists at '${phpfpm_conf}'.
- You should safely delete it before restoring this app."
-
-# Restore the app files
-sudo cp -a ./sources "$DESTDIR"
-
-# Create and restore the database
-ynh_mysql_create_db $dbname $dbuser $dbpass
-ynh_mysql_connect_as $dbuser $dbpass $dbname < ./dump.sql
-
-# Fix installation directories and permissions
-sudo mkdir -p "${DESTDIR}/logs" "${DESTDIR}/temp"
-sudo chown -R www-data: "$DESTDIR"
-
-# Restore configuration files
-sudo cp -a ./nginx.conf   "$nginx_conf"
-sudo cp -a ./php-fpm.conf "$phpfpm_conf"
-
-# Reload services
-sudo service php5-fpm restart || true
-sudo service nginx reload || true
+domain=$(ynh_app_setting_get $app domain)
+path_url=$(ynh_app_setting_get $app path)
+final_path=$(ynh_app_setting_get $app final_path)
+db_name=$(ynh_app_setting_get $app db_name)
+
+#=================================================
+# CHECK IF THE APP CAN BE RESTORED
+#=================================================
+
+ynh_webpath_available $domain $path_url \
```

Comment:
Shouldn't we halso register the URL via `ynh_webpath_register`?

Reply:
Not in the restore script.
It's also for this reason they split the command in 2 differents, because you can't register again the domain.
If my memories are good...

Reply:
But, you can't restore if the app is already installed... so the URL isn't registered either, is it...? :frowning:

Reply:
So, I have to ask to ljf... He's just in front of me.

Reply:
So, I have some answers !
In fact, `ynh_webpath_available` will just check if the the path is currently available in the sso (or somewhere else ?).
But, not store the information in the settings.yml file.

During the restore process, the settings.yml file is restored at the beginning, but the sso will not be reload until the end of the script. So the sso still consider the path is available.
If you use `ynh_webpath_register`, the sso will be reload immediately, and the path is not longer available.

In conclusion, it's not needed to used `ynh_webpath_register`, because the settings file is already restored. But I think you can if you use it after `ynh_webpath_available`.

Reply:
OK, roger that.
So, for what reasons did the checkurl command get split-up?

Reply:
Because, it's better to check if the path is available. But the old command had the bad habit to fail because the path is already in the settings.yml file.
I don't know much more about this command though...

Reply:
> There is also bug #983 regarding the use of checkurl in the remove script of example_ynh. If you confirm this PR as an model, we'll have fix the example_ynh bug accordingly.

It's already fixed on my local copy. I made a lot of correction on example_ynh.